pkg/volume/portworx/portworx_util.go

```diff
@@ -72,12 +72,14 @@ func (util *portworxVolumeUtil) CreateVolume(p *portworxVolumeProvisioner) (stri
 	}
 
 	// Pass all parameters as volume labels for Portworx server-side processing
-	if len(p.options.Parameters) > 0 {
-		spec.VolumeLabels = p.options.Parameters
-	} else {
+	if spec.VolumeLabels == nil {
```

Comment:
this doesn't seem to protect against concurrent access... was the issue concurrency or assigning to a nil map?

Reply:
@liggitt  thanks for reviewing this.

Before this change, reference to `p.options.Parameters` (which is the shared map) was getting saved in `spec.VolumeLabels` and line 103 below updates this map. When multiple PVCs are getting provisioned, this causes concurrent map writes. The golang panic will say `fatal: concurrent map read and write`.

Another problem you can see from the below for loop is `p.options.Parameters` will start accumulate annotations of all PVCs sharing a StorageClass.

Let me know if you have further questions.

Reply:
I see, thanks